modules/appeals_api/app/workers/appeals_api/daily_stuck_records_report.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'sidekiq'
+require 'sidekiq/monitored_worker'
+
+class AppealsApi::DailyStuckRecordsReport
+  include Sidekiq::Worker
+  include Sidekiq::MonitoredWorker
+
+  # Only retry for ~8 hours since the job runs daily
+  sidekiq_options retry: 11, unique_for: 8.hours
+
+  def perform
+    return unless enabled?
+
+    params = find_stuck_record_data(AppealsApi::HigherLevelReview) + \
+             find_stuck_record_data(AppealsApi::NoticeOfDisagreement) + \
+             find_stuck_record_data(AppealsApi::SupplementalClaim)
+
+    return if params.empty?
+
+    AppealsApi::Slack::Messager.new(params, notification_type: :stuck_record).notify!
+  end
+
+  def enabled?
+    Flipper.enabled? :appeals_daily_stuck_records_report
+  end
+
+  def retry_limits_for_notification
+    [11]
+  end
+
+  def notify(retry_params)
+    AppealsApi::Slack::Messager.new(retry_params, notification_type: :error_retry).notify!
+  end
+
+  private
+
+  def find_stuck_record_data(klass)
+    data = []
+    klass.stuck_unsubmitted&.find_each do |record|
```

Comment:
Since you're only slicing for a subset of the record's fields, I'd argue you should add a `klass.select(*attrs)...` to only grab the needed data from the database. You should then have the benefit of only needing to call `record.to_h.merge(...)` in your block, instead of slicing.

Using a `select` in this case also has more benefits:
* reduces security exposure since it won't transfer - nor open the possibly to decrypt - the encrypted fields that aren't being used.
* reduces network transfer of data & in some cases db load - less work to collect fewer fields in general
* Shrinks the ActiveRecord object size, which is notoriously heavy. Not really a worry here, but it's a technique to help when you need it :)

Reply:
When I use `select` here, it doesn't seem to be working - I still get an instance of an Appeal with all attributes present in the body of the `find_each`, and I get a `NoMethodError` if I try to use `to_h` on it. Here's what I tried:

```ruby
    klass.stuck_unsubmitted&.select(:id, :status, :created_at).find_each do |record|
      data << record.to_h.merge({ record_type: klass.name.demodulize })
    end
```

I feel like I'm not understanding some aspect of ActiveRecord here - could this have something to do with chaining onto the scoped `when` in `stuck_unsubmitted`? Or maybe something else?

Reply:
Ok I pushed one more update after we talked in slack - I think things are looking better now